app/connectors/MarriageAllowanceDataConnector.scala

```diff
@@ -38,50 +42,129 @@ object MarriageAllowanceDataConnector extends MarriageAllowanceDataConnector wit
-    implicit val hc = createHeaderCarrier
+    Logger.error(s"Not able to parse the response received from DES with error ${extractValidationErrors(err)}")
+    Left(ResponseValidationError)
+  }
+
+  def findCitizen(nino: Nino)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[JsValue] = {
     val path = url(s"/marriage-allowance/citizen/${nino}")
-    httpGet.GET[JsValue](path)
+    httpGet.GET[JsValue](path)(implicitly, buildHeaderCarrier(hc), ec)
   }
 
-  def listRelationship(cid: Cid, includeHistoric: Boolean = true)(implicit ec: ExecutionContext): Future[JsValue] = {
-    implicit val hc = createHeaderCarrier
+  def listRelationship(cid: Cid, includeHistoric: Boolean = true)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[JsValue] = {
     val path = url(s"/marriage-allowance/citizen/${cid}/relationships?includeHistoric=${includeHistoric}")
-    httpGet.GET[JsValue](path)
+    httpGet.GET[JsValue](path)(implicitly, buildHeaderCarrier(hc), ec)
   }
 
-  def findRecipient(findRecipientRequest: FindRecipientRequest)(implicit ec: ExecutionContext): Future[JsValue] = {
-    implicit val hc = createHeaderCarrier
-    val query = s"surname=${utils.encodeQueryStringValue(findRecipientRequest.lastName)}&forename1=${utils.encodeQueryStringValue(findRecipientRequest.name)}&gender=${utils.encodeQueryStringValue(findRecipientRequest.gender.gender)}"
-    val nino = findRecipientRequest.nino.nino.replaceAll(" ", "")
-    val path = url(s"/marriage-allowance/citizen/${nino}/check?${query}")
-    httpGet.GET[JsValue](path)
+  def findRecipient(findRecipientRequest: FindRecipientRequest)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Either[DataRetrievalError, UserRecord]] = {
+
+    def evaluateCodes(findRecipientResponseDES: FindRecipientResponseDES): Either[DataRetrievalError, UserRecord] = {
+
+      (findRecipientResponseDES.returnCode, findRecipientResponseDES.reasonCode) match {
+        case(ProcessingOK, ProcessingOK) =>{
+          metrics.incrementSuccessCounter(ApiType.FindRecipient)
+          Right(UserRecord(findRecipientResponseDES.instanceIdentifier, findRecipientResponseDES.updateTimeStamp))
+        }
+        case codes @ (ErrorReturnCode, NinoNotFound) => {
+          val codedErrorResponse = FindRecipientCodedErrorResponse(codes._1, codes._2, "Nino not found and Nino not found in merge trail")
+          Logger.warn(codedErrorResponse.errorMessage)
+          metrics.incrementSuccessCounter(ApiType.FindRecipient)
+          Left(codedErrorResponse)
+        }
+        case codes @ (ErrorReturnCode, MultipleNinosInMergeTrail) => {
+          val codedErrorResponse = FindRecipientCodedErrorResponse(codes._1, codes._2, "Nino not found and Nino found in multiple merge trails")
+          Logger.warn(codedErrorResponse.errorMessage)
+          metrics.incrementSuccessCounter(ApiType.FindRecipient)
+          Left(codedErrorResponse)
+        }
+        case codes @ (ErrorReturnCode, ConfidenceCheck) => {
```

Comment:
I cant find unit tests for this branch of code under the Coded error responses tests